### ISSUE TITLE
stuck when training using dist_train

### ISSUE THREAD

**Issue by banxia1994** [open-mmlab/mmdetection#1979]
when run the script “ ./tools/dist_train.sh ./configs/mask_rcnn_r50_fpn_1x.py 4 ”,meet the problem
2020-01-14 09:01:53,683 - INFO - workflow: [('train', 1)], max: 12 epochs

Stuck here for hours, please help

**Comment by jiangzhengkai** [comment]
@banxia1994  have you solved  the problem?

**Comment by banxia1994** [comment]
> @banxia1994 have you solved the problem?

not yet

**Comment by Gr4ph1c** [comment]
@banxia1994 @jiangzhengkai 
in bbox_head.py 120lines
add losses['loss_bbox'] = torch.tensor(0.).to(bbox_pred.device) before if pos_inds.any(): 
this bug will be solved

**Comment by jiangzhengkai** [comment]
@Gr4ph1c  Actually, you are right. The problem can be solved by Initializing the loss.

**Comment by yan-roo** [comment]
Same problem but not useful here

**Comment by banxia1994** [comment]
thx

**Comment by leemengwei** [comment]
> 
> 
> @banxia1994 @jiangzhengkai
> in bbox_head.py 120lines
> add losses['loss_bbox'] = torch.tensor(0.).to(bbox_pred.device) before if pos_inds.any():
> this bug will be solved

Thanks! But what is the exact reason for GPU being stucked with utility 100% erverlasting?  Is't something of MultiGPU conrespondce? (as it works fine with single GPU train)

**Comment by yangninghua** [comment]
> > @banxia1994 @jiangzhengkai
> > in bbox_head.py 120lines
> > add losses['loss_bbox'] = torch.tensor(0.).to(bbox_pred.device) before if pos_inds.any():
> > this bug will be solved
> 
> Thanks! But what is the exact reason for GPU being stucked with utility 100% erverlasting? Is't something of MultiGPU conrespondce? (as it works fine with single GPU train)

I met it, too. Did you solve it when I was single machine and multi card

**Comment by Gr4ph1c** [comment]
> > > @banxia1994 @jiangzhengkai
> > > in bbox_head.py 120lines
> > > add losses['loss_bbox'] = torch.tensor(0.).to(bbox_pred.device) before if pos_inds.any():
> > > this bug will be solved
> > 
> > 
> > Thanks! But what is the exact reason for GPU being stucked with utility 100% erverlasting? Is't something of MultiGPU conrespondce? (as it works fine with single GPU train)
> 
> I met it, too. Did you solve it when I was single machine and multi card

my method is also work for single machine with multi cards

**Comment by Gr4ph1c** [comment]
> > @banxia1994 @jiangzhengkai
> > in bbox_head.py 120lines
> > add losses['loss_bbox'] = torch.tensor(0.).to(bbox_pred.device) before if pos_inds.any():
> > this bug will be solved
> 
> Thanks! But what is the exact reason for GPU being stucked with utility 100% erverlasting? Is't something of MultiGPU conrespondce? (as it works fine with single GPU train)

I think it's a multithreading problem

**Comment by yangninghua** [comment]
Hello Gr4ph1c, is this the format?  @Gr4ph1c 
```
        if bbox_pred is not None:
            pos_inds = labels > 0
            losses['loss_bbox'] = torch.tensor(0.).to(bbox_pred.device)
            if pos_inds.any():
                if self.reg_class_agnostic:
                    pos_bbox_pred = bbox_pred.view(bbox_pred.size(0),
                                                   4)[pos_inds]
                else:
                    pos_bbox_pred = bbox_pred.view(bbox_pred.size(0), -1,
                                                   4)[pos_inds,
                                                      labels[pos_inds]]
```

**Comment by yangninghua** [comment]
He doesn't seem to work for me.
https://github.com/open-mmlab/mmdetection/tree/v1.0.0

**Comment by Gr4ph1c** [comment]
> He doesn't seem to work for me.
> https://github.com/open-mmlab/mmdetection/tree/v1.0.0

my version is v1.0rc1 not the latest，

> Hello Gr4ph1c, is this the format? @Gr4ph1c
> 
> ```
>         if bbox_pred is not None:
>             pos_inds = labels > 0
>             losses['loss_bbox'] = torch.tensor(0.).to(bbox_pred.device)
>             if pos_inds.any():
>                 if self.reg_class_agnostic:
>                     pos_bbox_pred = bbox_pred.view(bbox_pred.size(0),
>                                                    4)[pos_inds]
>                 else:
>                     pos_bbox_pred = bbox_pred.view(bbox_pred.size(0), -1,
>                                                    4)[pos_inds,
>                                                       labels[pos_inds]]
> ```

yes

**Comment by Gr4ph1c** [comment]
> He doesn't seem to work for me.
> https://github.com/open-mmlab/mmdetection/tree/v1.0.0

if this method does not work, i think you can do some debug about the losses e.g. printing the information about losses

**Comment by ZlodeiBaal** [comment]
Same issue here.
But i found solution for me - pytorch/pytorch#33296
import cv2
At the beginning of the program ( in train.py ), first import cv2, before import torch. Seem like pytorch internal problem.
